superset/db_engine_specs/presto.py

```diff
@@ -910,22 +911,26 @@ def _partition_query(cls, table_name, limit=0, order_by=None, filters=None):
     @classmethod
     def where_latest_partition(cls, table_name, schema, database, qry, columns=None):
         try:
-            col_name, value = cls.latest_partition(
+            col_names, values = cls.latest_partition(
                 table_name, schema, database, show_first=True
             )
         except Exception:
             # table is not partitioned
             return False
-        if value is not None:
-            for c in columns:
-                if c.get("name") == col_name:
-                    return qry.where(Column(col_name) == value)
+        if values is not None:
+            is_partitioned = False
+            for i, col_name in enumerate(col_names):
+                if (column.get("name") == col_name for column in columns):
+                    is_partitioned = True
+                    qry = qry.where(Column(col_name) == values[i])
+            if is_partitioned:
+                return qry
```

Comment:
You can optimize this whole block a little bit by storing the column names in a set:

```python
...
if values is None:
   return False

column_names = {column.get('name') for column in columns or []}
for col_name, value in zip(col_names, values):
    if col_name in column_names:
        qry = qry.where(Column(col_name) == value)
return qry
```

Also, do we need the `if is_partitioned` check? Is there a case where we get the latest partitions from `cls.latest_partition()` but none of the returned columns are in `columns`?

And what happens when `columns` is `None`? Can that ever happen? (I added a `columns or []` in my example to prevent errors when it's `None`.)

Reply:
I don't really think or understand why we need to return false in this function but it appears that the original function would return the query if it was modified and false otherwise. So that is why I added the is_partitioned check to maintain this thought. 

Although there is only one place we call where_latest_partition and that part of the code will just use the qry if where_latest_partition returns false.

```
        if latest_partition:
            partition_query = cls.where_latest_partition(
                table_name, schema, my_db, qry, columns=cols
            )
            if partition_query != False:  # noqa
                qry = partition_query
```

Do you know the history or background of why it is important to false for unmodified queries?

For your other questions, ideally the latest partitions should be in `columns` and I don't believe that `columns` should be `None` but I supposed those were extra checks added in the past.

Reply:
I took your feedback though to optimize that code block though.

Reply:
> Do you know the history or background of why it is important to (return) false for unmodified queries?

I can't think of a good reason, especially because it's only used in that place. I think it would be clearer if `where_latest_partition` always returned a query, and it would simplify the logic.

Reply:
Totally agree. Cool I have updated the PR with your feedback, including this one.

Reply:
Actually, I think there's one advantage... returning `False` indicates that the table is *not partitioned*, either because (1) `latest_partition()` raised an exception or (2) because the partition is not present in the list of columns (which should never happen, AFAICT).

Since condition (2) should never happen, I think it's fine to `return qry` at the end of the block, since it will have been modified.